plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -74,6 +77,31 @@ data ScriptDecodeError
 
 makeClassyPrisms ''ScriptDecodeError
 
+instance Pretty ScriptDecodeError where
+  pretty = \case
+    CBORDeserialiseError e ->
+      "Failed to deserialise a script:" <+> pretty e
+    RemainderError bs ->
+      "Script was successfully deserialised, but"
+        <+> pretty (BSL.length bs)
+        <+> "more bytes were encountered after the script's position."
+    LedgerLanguageNotAvailableError{..} ->
+      "You are attempting to use a"
+        <+> pretty sdeAffectedLang
+        <+> "script which is not yet supported by the current major protocol version"
+        <+> pretty sdeThisPv <> "."
+        <+> "The major protocol version that introduces this language is"
+        <+> pretty sdeIntroPv <> "."
+        <+> "To learn more about Plutus Core language versions, see \
+        \https://plutus.readthedocs.io/en/latest/explanations/language-versions.html."
```

Comment:
I wouldn't link to this page. The error message is clear enough, and this page is outdated, and is not exactly an explanation of why this error happened: it doesn't even mention "protocol version".

By the way, to learn about the different versions, take a look at [this doc](https://docs.google.com/document/d/1VH1ji6V0zIfywk8uy1a8HwCXcx9Y5fX0AetsgPOfrHA) (which we should publish on the doc site) if you haven't.

Reply:
Sure, will remove the link. Thanks for the doc reference.